### PR TITLE
Add header to get artifact attribute request

### PR DESCRIPTION
**Pull request by artsiomtserashkovich**
Add header to get artifact attribute request to identify which version of artifact flow client supports

## Before submitting checklist

- [x] Did you **update the CHANGELOG**? (not for test updates, internal changes/refactors or CI/CD setup)
- [x] Did you **ask the docs owner** to review all the user-facing changes?

### CODE REVIEW

File: src/neptune/internal/backends/hosted_neptune_backend.py [hunk 1]
```diff
@@ -1103,3 +1104,15 @@ def _get_all_items(get_portion, step):
             previous_items = get_portion(limit=step, offset=len(items))
             items += previous_items
         return items
+
+    @staticmethod
+    def _add_artifact_version_to_request_params(default_request_params: Dict) -> Dict:
```
Comment (Raalsky) [inline src/neptune/internal/backends/hosted_neptune_backend.py:1109]: How about moving it to `hosted_artifact_operations` and making it "global" across all artifacts related requests and making `2` some constant?

File: src/neptune/internal/backends/hosted_neptune_backend.py [hunk 2]
```diff
@@ -834,10 +834,11 @@ def get_datetime_attribute(
...
     ) -> ArtifactAttribute:
+        requests_params = self._add_artifact_version_to_request_params(DEFAULT_REQUEST_KWARGS)
         params = {
             "experimentId": container_id,
             "attribute": path_to_str(path),
-            **DEFAULT_REQUEST_KWARGS,
+            **requests_params,
         }
         try:
             result = self.leaderboard_client.api.getArtifactAttribute(**params).response().result
```
Comment (Raalsky) [inline src/neptune/internal/backends/hosted_neptune_backend.py:844]: How about delegating this method to `hosted_artifact_operations`? I mean moving the content to some method that accepts swagger API object.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,8 @@
+## [UNRELEASED] neptune 1.6.2
+### Changes
```
Comment (Raalsky) [inline CHANGELOG.md:2]: Can we make a newline between the header and the `### Changes`? I'm really sorry for asking for that :D

File: src/neptune/internal/backends/hosted_neptune_backend.py (hunk 2 above)
Comment (artsiomtserashkovich) [reply src/neptune/internal/backends/hosted_neptune_backend.py:844]: Do you mean to add `X-Neptune-Artifact-Api-Version` to all requests within `hosted_artifact_operations`? 
Initially, I did it like that, but later notice we used it only with `get_artifact_attribute`.

Are there any plans for that header in the future, that I'm not aware of?

File: src/neptune/internal/backends/hosted_neptune_backend.py (hunk 1 above)
Comment (artsiomtserashkovich) [reply src/neptune/internal/backends/hosted_neptune_backend.py:1109]: Do you mean to add `X-Neptune-Artifact-Api-Version` to all requests within `hosted_artifact_operations`? 
Initially, I did it like that, but later notice we used it only with `get_artifact_attribute`.

Are there any plans for that header in the future, that I'm not aware of?

Comment (Raalsky) [reply src/neptune/internal/backends/hosted_neptune_backend.py:1109]: > Do you mean to add X-Neptune-Artifact-Api-Version to all requests within hosted_artifact_operations?
Initially, I did it like that, but later notice we used it only with get_artifact_attribute

Yes and you're correct that for now it's only needed for `get_artifact_attribute`. But it's mostly to simplify some future feature decompositions.

> Are there any plans for that header in the future, that I'm not aware of?

No, or at least not yet. We're probably going to change the Artifacts API completely.

Comment (artsiomtserashkovich) [reply src/neptune/internal/backends/hosted_neptune_backend.py]: Done

File: src/neptune/internal/backends/hosted_neptune_backend.py (hunk 2 above)
Comment (artsiomtserashkovich) [reply src/neptune/internal/backends/hosted_neptune_backend.py]: Done. Also moved `list_artifact_files` method implementation.

**Review by Raalsky — APPROVED** [review]

**Pull request merged**